src/main/java/service/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package service.impl;
+
+import db.Storage;
+import model.FruitTransaction;
+import service.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private static final FruitBalanceCheckService FRUIT_BALANCE_CHECK_SERVICE

```

Comment:
reference-type fields should not be constants
```suggestion
    private final FruitBalanceCheckService fruitBalanceCheckService 
```